src/python/pants/backend/python/target_types.py

```diff
@@ -808,9 +808,17 @@ def pex_binary_addresses(self) -> Addresses:
 @dataclass(frozen=True)
 class ResolvePythonDistributionEntryPointsRequest:
     """Looks at the entry points to see if it is a setuptools entry point, or a BUILD target address
-    that should be resolved into a setuptools entry point."""
+    that should be resolved into a setuptools entry point.
 
-    entry_points_field: PythonDistributionEntryPointsField
+    If the `entry_points_field` is present, inspect the specified entry points.
+    If the `provides_field` is present, inspect the `provides_field.kwargs["entry_points"]`.
+
+    This is to support inspecting one or the other depending on use case, using the same
+    logic for resolving pex_binary addresses etc.
+    """
+
+    entry_points_field: Optional[PythonDistributionEntryPointsField] = None
+    provides_field: Optional[PythonProvidesField] = None
```

Comment:
I extended the support for fields to look at for resolving entry points, so we get the same logic applied consistently across all different ways to provide entry points.

Reply:
To be extra defensive, you could add a `def __post__init()` that raises `AssertionError` if both are unspecified.